query/search/search.go

```diff
@@ -16,18 +16,20 @@ package search
 
 import (
 	"github.com/tigrisdata/tigris/query/filter"
+	"github.com/tigrisdata/tigris/query/read"
 )
 
 const (
 	all = "*"
 )
 
 type Query struct {
-	Q        string
-	Fields   []string
-	Facets   Facets
-	PageSize int
-	WrappedF *filter.WrappedFilter
+	Q              string
+	SearchFields   []string
+	Facets         Facets
+	PageSize       int
+	WrappedF       *filter.WrappedFilter
+	FieldSelection *read.FieldFactory
```

Comment:
just have a simple list, no need to have it as a map because I don't see in the future if we need to add any functionality inside the fields object